test/src/kbasesearchengine/test/parse/ObjectParserTest.java

```diff
@@ -0,0 +1,256 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.commons.io.FileUtils;
+import org.ini4j.Ini;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.CharStreams;
+
+import kbasesearchengine.common.GUID;
+import kbasesearchengine.common.ObjectJsonPath;
+import kbasesearchengine.events.handler.SourceData;
+import kbasesearchengine.parse.IdMapper;
+import kbasesearchengine.parse.ObjectParser;
+import kbasesearchengine.parse.SimpleIdConsumer;
+import kbasesearchengine.parse.SimpleSubObjectConsumer;
+import kbasesearchengine.parse.SubObjectConsumer;
+import kbasesearchengine.system.ObjectTypeParsingRules;
+import us.kbase.common.service.UObject;
+
+public class ObjectParserTest {
+
+	private static Map<String, String> config = null;
+	private static Path scratch;
+	private static ObjectParser impl = null;
+
+	@BeforeClass
+	public static void init() throws Exception {
+		// config loading
+		String configFilePath = System.getenv("KB_DEPLOYMENT_CONFIG");
+		if (configFilePath == null) {
+			configFilePath = System.getProperty("KB_DEPLOYMENT_CONFIG");
+		}
+		File deploy = new File(configFilePath);
+		Ini ini = new Ini(deploy);
+		config = ini.get("KBaseSearchEngine");
+		scratch = Paths.get(config.get("scratch"), "test_object_parser");
+
+		impl = new ObjectParser();
+	}
+
+	@AfterClass
+	public static void teardown() throws Exception {
+		FileUtils.deleteDirectory(scratch.toFile());
+	}
+
+	@Test
+	public void prepareTempFileTest() throws Exception {
+
+		File tempDir = new File(scratch.toString(), UUID.randomUUID().toString());
+		assertEquals(false, tempDir.exists());
+
+		@SuppressWarnings("static-access")
+		File tmpFile = impl.prepareTempFile(tempDir);
+		assertEquals(true, tempDir.exists());
+		assertEquals(true, tmpFile.exists());
+		
+		assertThat(tmpFile.toString(), both(containsString("ws_srv_response_")).and(containsString(".json")));
+	}
+
+	@SuppressWarnings("static-access")
+	@Test
+	public void extractParentFragmentGenomeFeatureTest() throws Exception {
+
+		String jsonResource = "genome01";
+		String type = "GenomeFeature";
+
+		String parentJson = this.extractParentFragment(type, jsonResource);
+		String expectedParentJson = "{\"domain\":\"B\",\"scientific_name\":\"Shewanella\",\"assembly_ref\":\"1/2/1\"}";
+		assertEquals(parentJson, expectedParentJson);
+	}
+
+	@SuppressWarnings("static-access")
+	@Test
+	public void extractParentFragmentGenomeTest() throws Exception {
+
+		String jsonResource = "genome01";
+		String type = "Genome";
+
+		String parentJson = this.extractParentFragment(type, jsonResource);
+		assertNull(parentJson); // Genome.json doesn't have "from-parent" path
+	}
+
+	@SuppressWarnings("unchecked")
```

Comment:
Rather than marking an entire method as unchecked, just mark the variable(s).

Reply:
👍